src/Adaptivity/RefinementRules.jl

```diff
@@ -0,0 +1,133 @@
+
+struct RefinementRule{T,A,B,C,D,E}
```

Comment:
Might it make sense to add a trait encode the refinement strategy e.g., to allow for dispatching in `refine`

Reply:
Addressed in new commits.